### DISCUSSION TITLE
Interested in Contributing

### DISCUSSION THREAD

**Discussion by t-botea** [DirtyRacer1337/Jellyfin.Plugin.PhoenixAdult#100] [General]
This project is great and there's tons of sites out there with many others requesting new sites every day. I'm interesting in contributing and adding additional site supports.  What's the best way to contribute new sites with regards to guidelines and/or coding style? My thought is to go through the git history and look at supports for new sites i.e. `c0ed6bec`

I'm also interested in adding test suites to the project. It would make it easier to verify supporting sites still behave as expected. Have this been thought about? I took a quick look into the source code and right now the architecture makes it difficult to add tests and mocks (static objects that depend on other static objects that depend on static methods, so on so on).

**Comment by DirtyRacer1337** [comment]
Code example can be looked at #79
For codestyle exist `.editorconfig` and two plugins `Microsoft.CodeAnalysis.NetAnalyzers` and `StyleCop.Analyzers`

I thinked about tests, but I got problem with plugin initialization, splitting plugin to provider for Jellyfin/Emby and data graber can solve this.
And yeah architecture is bad, it need lot improvements

**Reply by t-botea** [reply]
One step at a time! 

In terms of validating work, how have you set up your workflow to assert input? Currently, if I want to test any changes I would build the `.dll`, inject into my Emby Server, then start searching. This is not efficient and extremely slow. I'm wondering what you do that could make work faster.

**Reply by DirtyRacer1337** [reply]
I use jellyfin, just load debug build and can see errors in logs. Splitting plugin to two modules can solve this, just need to find time to do it.

**Reply by t-botea** [reply]
Just to be clear, loading debug build means changing the log level in `logging.default.json` to `Debug`. Is there anything else I need to do? And for checking errors in logs, you're looking in the `Inspect` window under `Console` panel?

**Reply by DirtyRacer1337** [reply]
I made like this, and messages appear in regular jelly logs

`logging.default.json`
```json
{
  "Serilog": {
    "MinimumLevel": {
      "Default": "Information",
      "Override": {
        "Microsoft": "Warning",
        "System": "Warning",
        "PhoenixAdult": "Debug",
        "ThePornDB": "Debug"
      }
    }
  }
}
```

I don't know how to load debug build to Emby, better use Jellyfin for it, here all files you need in plugin folder
```
FlareSolverrSharp.dll
HtmlAgilityPack.dll
HttpClient.Caching.dll
MihaZupan.HttpToSocks5Proxy.dll
PhoenixAdult.dll
PhoenixAdult.pdb
```

And don't forget it should be debug build, not release

**Reply by t-botea** [reply]
Yep, I'm using Jellyfin now. Trying to set up my dev environment so apologies for questions. I've worked in mobile development and management for years so this is new (and exciting) for me.

I've tried to load `debug` build and restarting my Jellyfin server. I see it under my `Plugins` but it says it's unsupported, any reason why?

<img src="https://user-images.githubusercontent.com/92008828/137042340-bc80fe43-81f5-4f6a-b5c5-f694c8274c76.PNG" width=700/>

**Reply by t-botea** [reply]
And to make sure my development workflow is the same as yours, after making changes, you build the `.dll`, import to `plugins`, and verify changes inside Jellyfin Web UI?

**Reply by DirtyRacer1337** [reply]
I think you should put all dll's inside subfolder, like `PhoenixAdult`

Yeah just drop `PhoenixAdult.dll` and check how it work as user should do it
